### PR TITLE
PxGraf integration related fixes

### PR DESCRIPTION
**Pull request by sakari-malkki**
Minor fixes for issues that were found during PxGraf integration process

### CODE REVIEW

File: Px.Utils/ModelBuilders/MatrixMetadataBuilder.cs [hunk 1]
```diff
@@ -91,12 +91,12 @@ private MatrixMetadata BuildFromentries(Dictionary<MetadataEntryKey, string> ent
             string stubKey = _pxFileSyntaxConf.Tokens.KeyWords.StubDimensions;
             IEnumerable<MultilanguageString> stubDimensionNames = TryGetAndRemoveProperty(entries, stubKey, langs, out MetaProperty? maybeStub)
                 ? maybeStub.ValueAsListOfMultilanguageStrings(langs.DefaultLanguage, _listSeparator, _stringDelimeter)
-                : throw new ArgumentException("Stub variable names not found in metadata");
```
Comment (jsaarimaa) [inline Px.Utils/ModelBuilders/MatrixMetadataBuilder.cs:94]: STUB and HEADING keywords are mandatory, if they are missing an exception should be thrown and handled in the calling function.

File: Px.Utils/Models/Metadata/ExtensionMethods/MatrixMetadataExtensions.cs [hunk 2]
```diff
@@ -1,19 +1,44 @@
...
             return timeDimension ?? throw new InvalidOperationException("Time dimension not found in metadata");
         }
+
+        /// <summary>
+        /// Tries to get the content dimension from the metadata.
+        /// </summary>
+        /// <param name="contentDimension">The content dimension if found.</param>
+        /// <returns>True if the content dimension is found, false otherwise.</returns>
+        /// <remarks>Use this method when missing the content dimension is not an exceptional case.</remarks>
+        public static bool TryGetContentDimension(this IReadOnlyMatrixMetadata metadata, out ContentDimension contentDimension)
```
Comment (jsaarimaa) [inline Px.Utils/Models/Metadata/ExtensionMethods/MatrixMetadataExtensions.cs:38]: The content dimension is mandatory, if it is missing the exception should be handled in the calling function.

File: Px.Utils/PxFile/Metadata/PxFileMetadataReader.cs [hunk 3]
```diff
@@ -138,7 +134,7 @@ public async IAsyncEnumerable<KeyValuePair<string, string>> ReadMetadataAsync(
                             throw new InvalidPxFileMetadataException($"Unexpected character '{parsingBuffer[i]}' found at position {i}.");
                         }
 
-                        if (parsingBuffer[i] == nextDelimeter)
+                        if (parsingBuffer[i] == keywordSeperator || parsingBuffer[i] == sectionSeparator)
```
Comment (jsaarimaa) [inline Px.Utils/PxFile/Metadata/PxFileMetadataReader.cs:137]: This changes the behaviour of the function. Now both separator types are accepted all the time. I would double check if this is actually related to the original issue.

File: Px.Utils/ModelBuilders/MatrixMetadataBuilder.cs (hunk 1 above)
Comment (sakari-malkki) [reply Px.Utils/ModelBuilders/MatrixMetadataBuilder.cs]: Change reverted

File: Px.Utils/PxFile/Metadata/PxFileMetadataReader.cs (hunk 3 above)
Comment (sakari-malkki) [reply Px.Utils/PxFile/Metadata/PxFileMetadataReader.cs]: Change reverted

File: Px.Utils/Models/Metadata/ExtensionMethods/MatrixMetadataExtensions.cs (hunk 2 above)
Comment (sakari-malkki) [reply Px.Utils/Models/Metadata/ExtensionMethods/MatrixMetadataExtensions.cs]: Change reverted

**Review by jsaarimaa — APPROVED** [review]

**Pull request merged**